### PR TITLE
[torch-th] fix arm64-osx build

### PR DESCRIPTION
**Pull request by autoantwort**

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@autoantwort , Thanks for your pr, could you please consider migrating them to the new functions:
vcpkg_install_cmake -> vcpkg_cmake_install (from port vcpkg-cmake)
vcpkg_configure_cmake -> vcpkg_cmake_configure (Please remove the option PREFER_NINJA) (from port vcpkg-cmake)

In the ports that use the new function, you have to add the corresponding dependencies:
````
{
  "name": "vcpkg-cmake",
  "host": true
}
````

**Comment by JonLiu1993** [comment]
Please add `license ` in vcpkg.json.

File: ports/torch-th/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,14 @@
 {
   "name": "torch-th",
-  "version-string": "2019-04-19",
-  "port-version": 3,
+  "version-date": "2019-04-19",
+  "port-version": 4,
   "description": "Torch's TH library",
-  "homepage": "https://github.com/torch/torch7"
+  "homepage": "https://github.com/torch/torch7",
+  "license": "BSD-3-Clause",
```
Comment (JonLiu1993) [inline ports/torch-th/vcpkg.json:7]: https://github.com/torch/torch7/blob/master/COPYRIGHT.txt

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Submitted upstream: https://github.com/torch/torch7/pull/1244

Upstream seems dead and this change is obvious. Thanks!

**Pull request merged**